core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -103,16 +103,32 @@ class SnappyParser(session: SnappySession)
     var index = 0
     val len = s.length
     // use double if ending with D/d, float for F/f and long for L/l
+
     s.charAt(len - 1) match {
       case 'D' | 'd' =>
-        return newTokenizedLiteral(
-          java.lang.Double.parseDouble(s.substring(0, len - 1)), DoubleType)
-      case 'F' | 'f' =>
+        if (s.length > 2) {
+          s.charAt(len - 2) match {
+            case 'B' | 'b' => toDecimalLiteral(s.substring(0, len - 2), checkExactNumeric = false)
+            case c if (Character.isDigit(c)) => return newTokenizedLiteral(
+              java.lang.Double.parseDouble(s.substring(0, len - 1)), DoubleType)
+            case _ => throw new ParseException(s"Found non numeric token $s")
+          }
+        } else {
+          return newTokenizedLiteral(
+            java.lang.Double.parseDouble(s.substring(0, len - 1)), DoubleType)
+        }
+      case 'F' | 'f' => if (Character.isDigit(s.charAt(len - 2))) {
         return newTokenizedLiteral(
           java.lang.Float.parseFloat(s.substring(0, len - 1)), FloatType)
-      case 'L' | 'l' =>
+      } else {
+        throw new ParseException(s"Found non numeric token $s")
+      }
+      case 'L' | 'l' => if (Character.isDigit(s.charAt(len - 2))) {
```

Comment:
To be safe might be good to add length check too here and the case before: if (s.length > 2 && Character.isDigit(...))

Reply:
Is it needed? Because I suppose the parser rule itself would ensure atleast one number followed by suffix, so I deliberately avoided the check.